sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1573,6 +1586,27 @@ && runnerPrefersLegacyRead(input.getPipeline().getOptions()))) {
       return input.apply(new ReadFromKafkaViaSDF<>(this, keyCoder, valueCoder));
     }
 
+    private void checkRedistributeConfiguration() {
+      if (getRedistributeNumKeys() == 0 && isRedistributed()) {
+        LOG.warn(
+            "withRedistribute without withRedistributeNumKeys will create a key per record, which is sub-optimal for most use cases.");
+      }
+      if (isAllowDuplicates()) {
+        checkState(
```

Comment:
was previously a log warn

Reply:
I was trying to make the Kafka read config checking more consistent, to align with the direction we were going in with the offset deduplication config. I've changed it back to a warning in case that is less friction for customer upgrades and consistency with prior impl. If we want to change it back to a check, perhaps we can handle it on a separate change than this one.